blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitScm.java

```diff
@@ -90,60 +215,127 @@ public HttpResponse validateAndCreate(@JsonBody JSONObject request) {
                 } else {
                     return HttpResponses.errorJSON("No repository found for: " + fullName);
                 }
-            } catch(JSONException e) {
+            } catch (JSONException e) {
                 return HttpResponses.errorJSON("No repositoryUrl or pipeline.fullName specified in request.");
-            } catch(RuntimeException e) {
+            } catch (RuntimeException e) {
                 return HttpResponses.errorWithoutStack(ServiceException.INTERNAL_SERVER_ERROR, e.getMessage());
             }
         }
 
+        // --[ Grab user ]-------------------------------------------------------------------------------------
+
+        User user = User.current();
+        if (user == null) {
+            throw new ServiceException.UnauthorizedException("Not authenticated");
+        }
+
+        // --[ Get credential id from request or create from repo url ]----------------------------------------
+
+        String credentialId = null;
+
+        if (request.has("credentialId")) {
+            credentialId = request.getString("credentialId");
+        }
+
+        if (credentialId == null) {
+            credentialId = makeCredentialId(repositoryUrl);
+        }
+
+        if (credentialId == null) {
+            // Still null? Must be a bad repoURL
+            throw new ServiceException.BadRequestException("Invalid URL \"" + repositoryUrl + "\"");
+        }
+
+        // --[ Load or create credentials ]--------------------------------------------------------------------
+
+        // Create new is only for username + password
+        if (request.has("userName") || request.has("password")) {
```

Comment:
Should it not be &&, what condition it should be  either or?

Reply:
I just want to make sure you get an error if you're missing one, rather than silently pretending it wasn't attempted.